Added info about Open forms from other bindings, and updated bindings requirements. (#932)
mattermost/mattermost-developer-documentation: site/content/integrate/apps/bindings/_index.md

### `/post_menu` bindings

| Name       | Type   | Description                                                                                                       |
| :--------- | :----- | :---------------------------------------------------------------------------------------------------------------- |
| `location` | string | Name of this location. The whole path of locations will be added in the context. Must be unique in its level.     |
| `icon`     | string | (Optional) Either a fully-qualified URL, or a path for an app's static asset.                                     |
| `label`    | string | (Optional) Text to show in the item. Defaults to location. Must be unique in its level.                           |

> | `call`     | Call   | (Optional) Call to perform. You must provide a Call if there is no Form, or the Form itself does not have a Call. |

Reviewer: ```suggestion
| `call`     | Call   | (Optional) Call to perform. You must provide a call if there is no form, or the form itself does not have a call. |
```

> | `form`     | Form   | (Optional) Modal form to open. You must provide a Form with a Call if there is no Call defined in the Binding.    |

Reviewer: ```suggestion
| `form`     | Form   | (Optional) Modal form to open. You must provide a form with a call if there is no call defined in the binding.    |
```

> The call for these bindings will include in the context the user ID, the post ID, the root post ID if any, the channel ID and the team ID.

Reviewer: ```suggestion
The call for these bindings will include in the context the user ID, the post ID, the root post ID if any, the channel ID, and the team ID.
```

### `/channel_header` bindings

| Name       | Type   | Description                                                                                                                 |
| :--------- | :----- | :-------------------------------------------------------------------------------------------------------------------------- |
| `location` | string | Name of this location. The whole path of locations will be added in the context. Must be unique in its level.               |

> | `icon`     | string | (Optional / Webapp Required) Either a fully-qualified URL, or a path for an app's static asset.                             |

Reviewer: ```suggestion
| `icon`     | string | (Optional/Web App required) Either a fully-qualified URL, or a path for an app's static asset.                             |
```

Reviewer: Does this mean it's required in/by the Web App?

Author: This means that in general is optional, but it is required in the web app. Since there are only two different agents (webapp or mobile) we could phrase it as "optional in mobile, required in webapp", but I wanted to be "future proof".

Reviewer: I thought that might be the case but just wanted to check. I like the idea of future-proofing it! :)

| `label`    | string | (Optional) Text to show in the item on mobile and webapp collapsed view. Defaults to location. Must be unique in its level. |

> | `hint`     | string | (Optional / Webapp Required) Text to show in tooltip.                                                                       |

Reviewer: ```suggestion
| `hint`     | string | (Optional/Web App required) Text to show in tooltip.                                                                       |
```

> | `call`     | Call   | (Optional) Call to perform. You must provide a Call if there is no Form, or the Form itself does not have a Call.           |

Reviewer: ```suggestion
| `call`     | Call   | (Optional) Call to perform. You must provide a call if there is no form, or the form itself does not have a call.           |
```

> | `form`     | Form   | (Optional) Modal form to open. You must provide a Form with a Call if there is no Call defined in the Binding.              |

Reviewer: ```suggestion
| `form`     | Form   | (Optional) Modal form to open. You must provide a form with a call if there is no call defined in the binding.              |
```

A leaf command must include:

| Name          | Type   | Description                                                                                                                                  |
| :------------ | :----- | :------------------------------------------------------------------------------------------------------------------------------------------- |
| `location`    | string | Name of this location. The whole path of locations will be added in the context. Must be unique in its level.                                   |
| `label`       | string | The label to use to define the command. Cannot include spaces or tabs. Defaults to location. Must be unique in its level.                       |
| `hint`        | string | (Optional) Hint line on command autocomplete.                                                                                                   |
| `description` | string | (Optional) Description line on command autocomplete.                                                                                           |

> | `call`        | Call   | (Optional) Call to perform when executing the command. You must provide a Call if there is no Form, or the Form itself does not have a Call. |                                                                                                    |

Reviewer: ```suggestion
| `call`        | Call   | (Optional) Call to perform when executing the command. You must provide a call if there is no form, or the form itself does not have a call. |                                                                                                    |
```

| `form`        | Form   | (Optional) Form representing the parameters the command can receive. If no form is provided, a form call will be made to the specified call. |

> The context of the call for these bindings will include the user ID, the post ID, the root post ID (if any), the channel ID and the team ID. It will also include the raw command.

Reviewer: ```suggestion
The context of the call for these bindings will include the user ID, the post ID, the root post ID (if any), the channel ID, and the team ID. It will also include the raw command.
```
